examples/client.go

```diff
@@ -501,16 +501,71 @@ func main() {
 
 	case "drainHosts":
 		fmt.Println("Setting hosts to DRAINING")
-		if drainCandidates == "" {
+		if hostList == "" {
 			fmt.Println("No hosts specified to drain")
 			os.Exit(1)
 		}
-		hosts := strings.Split(drainCandidates, ",")
+		hosts := strings.Split(hostList, ",")
 		_, result, err := r.DrainHosts(hosts...)
 		if err != nil {
 			fmt.Printf("error: %+v\n", err.Error())
 			os.Exit(1)
 		}
+
+		// Monitor change to DRAINING and DRAINED mode
+		nontransitioned, err := monitor.HostMaintenance(
+			hosts,
+			[]aurora.MaintenanceMode{aurora.MaintenanceMode_DRAINED, aurora.MaintenanceMode_DRAINING},
+			5,
+			10)
+		if err != nil {
+
+			// Check whether the call was partially successful
+			if len(nontransitioned) != 0 {
+				fmt.Println("Partial success:")
+				for host, _ := range nontransitioned {
+					fmt.Printf("Host %s did not transtion into desired mode(s)\n", host)
+				}
+			}
+
+			fmt.Printf("error: %+v\n", err.Error())
+			os.Exit(1)
+		}
+
+		fmt.Print(result.String())
+
+	case "endMaintenance":
+		fmt.Println("Setting hosts to ACTIVE")
+		if hostList == "" {
+			fmt.Println("No hosts specified to drain")
+			os.Exit(1)
+		}
+		hosts := strings.Split(hostList, ",")
+		_, result, err := r.EndMaintenance(hosts...)
+		if err != nil {
+			fmt.Printf("error: %+v\n", err.Error())
+			os.Exit(1)
+		}
+
+		// Monitor change to DRAINING and DRAINED mode
+		nontransitioned, err := monitor.HostMaintenance(
```

Comment:
From the comment of HostMaintenance() it appears the returned map as all hosts and their statuses, not just the ones that have not transitioned to the desired state.

If that is true the name of variable "nontransitioned" is misleading.

Reply:
Good catch, this and below was due to a code change that I made on the API side from which I didn't update the client code.